python/ray/_raylet.pyx

```diff
@@ -5164,6 +5164,10 @@ cdef void async_callback(shared_ptr[CRayObject] obj,
 
         user_callback = <object>user_callback_ptr
         user_callback(result)
+    except Exception as e:
```

Comment:
we can remove `as e` here. Also, can we add a note and better an URL to some pages describing this potential bug on Python&Cython?